compiler/qsc_hir/src/ty.rs

```diff
@@ -50,6 +50,25 @@ pub enum Ty {
 impl Ty {
     /// The unit type.
     pub const UNIT: Self = Self::Tuple(Vec::new());
+
+    #[must_use]
+    pub fn with_package(&self, package: Option<PackageId>) -> Self {
+        match self {
+            Ty::Array(item) => Ty::Array(Box::new(item.with_package(package))),
+            Ty::Arrow(arrow) => Ty::Arrow(Box::new(arrow.with_package(package))),
+            Ty::Infer(infer) => Ty::Infer(*infer),
+            Ty::Param(param) => Ty::Param(*param),
+            Ty::Prim(prim) => Ty::Prim(*prim),
```

Comment:
Don't know much Rust, hence the question. Can you use the same trick for the three cases as you used before ("_ => *self,")

Reply:
Yes, actually I could!

Reply:
Actually, no, because each case needs to construct a different type.

Reply:
Ok, never mind. Came back again, and you are right. Each case was created an instance of the same type, so those can all be captured by a single case that clones. Good eye!